lib/instrumentation/modules/mongodb.js

```diff
@@ -28,6 +44,19 @@ module.exports = (mongodb, agent, { version, enabled }) => {
     const span = agent.startSpan(name, 'db', 'mongodb', 'query')
     if (span) {
       activeSpans.set(event.requestId, span)
+
+      // Per the following code it looks like "<hostname>:<port>" should be
+      // available via the `address` or `connectionId` field.
+      // https://github.com/mongodb/node-mongodb-native/blob/dd356f0ede/lib/core/connection/apm.js#L155-L169
+      const address = event.address || event.connectionId
+      let match
```

Comment:
Capturing an online conversation here -- both the docs and some spot checking of a port-less mongo connection (`mongoose.connect('mongodb://localhost/myapp', {useNewUrlParser: true});`) indicate that the address/connection ID will be normalized to include a port, even if the client is configured without one.  If somehow a portless address does end up in there, we won't hard fail as `exec` won't match and the conditional will fail.